code/controllers/Processes/aircurrents.dm

```diff
@@ -0,0 +1,480 @@
+#define AIR_CURRENTS_TEMPERATURE_SPEED 0.0001 //affect how fast the temperature spread using the air currents
+#define AIR_CURRENTS_PRESSUREPUSH_MULT 2 //due to how the air is averaged on the chain, the pressure difference is actually often much smaller than it should be, so we simply apply add a multiplier
+#define AIR_CURRENTS_SPACESIPHON_MULT 4 //if the source of the air current is next to a turf open to space, the current will "consume" AIR_CURRENTS_SPACESIPHON_MULT times the air the src used in its balance. This allow for accounting of the fact most of the air you add in the turf connected to space, end up continuing into space
+var/global/datum/controller/process/aircurrents/air_currents_master
+/datum/controller/process/aircurrents/var/list/active_air_currents[0] //our list of active air currents
+
+//uncomments to have a visual debug ingame of how the air currents act. Be warned this add a sizeable performance overhead
+//#define AIRCURRENTDEBUG 1
+#ifdef AIRCURRENTDEBUG
+
+var/obj/effect/overlay/debugsrc
+var/obj/effect/overlay/debug1
+var/obj/effect/overlay/debug2
+var/obj/effect/overlay/debug3
+var/obj/effect/overlay/debug4
+var/obj/effect/overlay/debug5
+var/obj/effect/overlay/debug6
+var/obj/effect/overlay/debug6p
+#endif
+/datum/controller/process/aircurrents/setup()
+	name = "air_currents"
+	schedule_interval = 20
+	start_delay = 4
+	air_currents_master = src
+#ifdef AIRCURRENTDEBUG
+	debugsrc = new /obj/effect/overlay()
+	debugsrc.icon = 'icons/misc/debug_group.dmi'
+	debugsrc.icon_state = ""
+	debugsrc.layer = FLY_LAYER
+	debugsrc.mouse_opacity = 0
+
+	debug1 = new /obj/effect/overlay()
+	debug1.icon = 'icons/misc/debug_rebuild.dmi'
+	debug1.icon_state = "1"
+	debug1.layer = FLY_LAYER
+	debug1.mouse_opacity = 0
+
+	debug2 = new /obj/effect/overlay()
+	debug2.icon = 'icons/misc/debug_rebuild.dmi'
+	debug2.icon_state = "2"
+	debug2.layer = FLY_LAYER
+	debug2.mouse_opacity = 0
+
+	debug3 = new /obj/effect/overlay()
+	debug3.icon = 'icons/misc/debug_rebuild.dmi'
+	debug3.icon_state = "3"
+	debug3.layer = FLY_LAYER
+	debug3.mouse_opacity = 0
+
+	debug4 = new /obj/effect/overlay()
+	debug4.icon = 'icons/misc/debug_rebuild.dmi'
+	debug4.icon_state = "4"
+	debug4.layer = FLY_LAYER
+	debug4.mouse_opacity = 0
+
+	debug5 = new /obj/effect/overlay()
+	debug5.icon = 'icons/misc/debug_rebuild.dmi'
+	debug5.icon_state = "5"
+	debug5.layer = FLY_LAYER
+	debug5.mouse_opacity = 0
+
+	debug6 = new /obj/effect/overlay()
+	debug6.icon = 'icons/misc/debug_rebuild.dmi'
+	debug6.icon_state = "6"
+	debug6.layer = FLY_LAYER
+	debug6.mouse_opacity = 0
+
+	debug6p = new /obj/effect/overlay()
+	debug6p.icon = 'icons/misc/debug_rebuild.dmi'
+	debug6p.icon_state = ""
+	debug6p.layer = FLY_LAYER
+	debug6p.mouse_opacity = 0
+#endif
+/datum/controller/process/aircurrents/statProcess()
+	..()
+	stat(null, "[active_air_currents.len] active")
+
+
+/datum/controller/process/aircurrents/doWork()
+	for (var/turf/simulated/T in active_air_currents)
```

Comment:
so if i understand what you want then. You want me to grab the for loop from doWork, put it in another proc called say process_allcurrents, call it from dowork, and add SCHECK into the for loop ?